pkgs/by-name/ls/lsp-ai/package.nix

```diff
@@ -0,0 +1,82 @@
+{
+  lib,
+  rustPlatform,
+  fetchFromGitHub,
+  stdenv,
+  darwin,
+  pkg-config,
+  cmake,
+  openssl,
+  zlib,
+  perl,
+  gitUpdater,
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "lsp-ai";
+  version = "0.7.0";
+  src = fetchFromGitHub {
+    owner = "SilasMarvin";
+    repo = "lsp-ai";
+    rev = "release/v${version}";
```

Comment:
```suggestion
    rev = "refs/tags/v${version}";
```

Reply:
This change has already been suggested. However, the tags don't get updated, only the release branches, so I believe the release branch is the best way to track patches up stream for a specific release cycle.

Reply:
Tags are updated automatically with each release. 

I suggested `release/v${version}` elsewhere and got the feedback "That's the URI for Github, but we need a git revision here" which is `refs/tags/v${version}`

Reply:
So in the name of stability what is recommended in this use case. Upstream tag does not match release branch, there is a cargo lock file version mismatch. I.e. the version release branch is ahead of the tagged release. If I revert to tags, the build will fail

Reply:
> Tags are updated automatically with each release. 
> 
> I suggested `release/v${version}` elsewhere and got the feedback "That's the URI for Github, but we need a git revision here" which is `refs/tags/v${version}`

That feedback is confusing because both are supported by the prefetch. Reproduciblity would come from the hashes no?

Reply:
Here's the specific feedback I received:

> refs/tags/* is correct, since this is a Git ref and not a GitHub URL.

Kindly follow the standard, please.

Reply:
To clarify why a branch is not an option: As soon as anything is pushed to the branch the hash will mismatch and therefore break the derivation.

If `refs/tags/<tag>` is unsuitable the only alternative is a revision hash.

Reply:
Okay, might have to use a revision hash. Upstream can also change tags which would also cause a mismatch. I'll look at the actual revision has and push an update.

Reply:
👋 I think there’s a bit of confusion here; GitHub release URLs are of the form `https://github.com/‹owner›/‹repo›/releases/‹tag›`, and so `releases/‹tag›` doesn’t work to refer to a Git tag in a fetch. However, what we’re talking about here is a *branch* named `release/v${version}` vs. a tag like `v${version}` or `refs/tags/v${version}` (the latter disambiguating when there is both a branch and a tag named `v${version}`, which is a thing that Git allows *for some reason*).

We cannot and will not ship a fetcher with a `rev` set to a branch name, because the fundamental property we want is that every fetch in Nixpkgs points to something reasonably expected to be immutable (a Git version tag or commit hash, a versioned tarball URL, etc.), whereas the whole purpose of branches is to push more commits to them (indeed, picking up those commits is the rationale being used here). Using a branch name here directly won’t work. The new commits won’t be picked up automatically; instead, there will simply be hash mismatches and general sadness. That is a blocking issue.

The two actual options are:

* Use a specific version tag and update it every time a new one is made.

* Pin the raw commit hash that is currently the HEAD of the corresponding `releases/…` branch, and periodically bump it.

There is no way to have Nixpkgs pick up a new version of a piece of software without a change being made to Nixpkgs that explicitly updates the pinned version and hash, and this is by design. If upstream recommends shipping the HEAD of their release branches and doesn’t cut tags in a timely manner, then we should manually pin and update commit hashes from that branch and use a version like `0.7.1-unstable-YYYY-MM-DD`. Otherwise, if tags are released frequently enough, we should simply use those. I don’t know enough about this project to determine what is best in this specific case and I don’t know what the upstream recommends, but I hope I’ve helped clear up some of the confusion here and clarify what the actual decision to be made is.

Reply:
> Here's the specific feedback I received:
> 
> > refs/tags/* is correct, since this is a Git ref and not a GitHub URL.
> 
> Kindly follow the standard, please.

I understand where the feedback/standard is coming from. I was merely pointing out it is still vague imo since the `src` fetcher used by so many packages is `fetchFromGitHub`. Seems pedantic because it's ultimately pulling from GitHub anyway ¯\_(ツ)_/¯

Thanks for the advice, I'll get an update pushed!

Reply:
Looks like my fix for tagging mismatch got merged, going to update to 0.7.1 to use the corrected [tag](https://github.com/SilasMarvin/lsp-ai/releases/tag/v0.7.1)

Reply:
@emilazy Your comment didn't come through before I posted my last two comments. That added context is super helpful. I think I have a much better understanding of the semantics of the task at hand. :)